exporter/exporterhelper/batch_sender.go

```diff
@@ -149,18 +154,24 @@ func (bs *batchSender) send(ctx context.Context, req Request) error {
 // sendMergeSplitBatch sends the request to the batch which may be split into multiple requests.
 func (bs *batchSender) sendMergeSplitBatch(ctx context.Context, req Request) error {
 	bs.mu.Lock()
-	bs.activeRequests.Add(1)
-	defer bs.activeRequests.Add(^uint64(0))
 
 	reqs, err := bs.mergeSplitFunc(ctx, bs.cfg.MaxSizeConfig, bs.activeBatch.request, req)
 	if err != nil || len(reqs) == 0 {
 		bs.mu.Unlock()
 		return err
 	}
+
+	bs.activeRequests.Add(1)
 	if len(reqs) == 1 || bs.activeBatch.request != nil {
 		bs.updateActiveBatch(ctx, reqs[0])
 		batch := bs.activeBatch
+		batch.requestsBlocked++
 		if bs.isActiveBatchReady() || len(reqs) > 1 {
+			if len(reqs) > 1 {
+				// there will be leftover requests and this request will be active until the function returns.
+				batch.requestsBlocked--
+				defer bs.activeRequests.Add(-1)
+			}
 			bs.exportActiveBatch()
```

Comment:
I think the logic is correct here. It can probably be simplified a bit more. Let me know WDYT.

```suggestion
	if len(reqs) == 1 {
		batch.requestsBlocked++
	} else {
		// if there was a split, we want to make sure that bs.activeRequests is released once all of the parts are sent instead of using batch.requestsBlocked
		defer bs.activeRequests.Add(-1)
	}
	if len(reqs) == 1 || bs.activeBatch.request != nil {
		bs.updateActiveBatch(ctx, reqs[0])
		batch := bs.activeBatch
		if bs.isActiveBatchReady() || len(reqs) > 1 {
			bs.exportActiveBatch()
```